### DISCUSSION TITLE
Migrating the GCP firestore DB to Local Postgresql DB

### DISCUSSION THREAD

**Discussion by teenwolf99** [gravitational/teleport#33107] [Q&A]
I have setup the teleport database in GCP fire store collection earlier, since now teleport supports PostgreSQL as backend in newer release. I wanted to know if its possible to migrate the DB collections from fire store to PostgreSQL. Does teleport provide any kind of DB migration tool to do this task.

**Comment by webvictim** [comment]
There is no migration tool provided for this.

You can, however, back up your backend data from Firestore using the backup and restore process documented here (https://goteleport.com/docs/management/operations/backup-restore/#migrating-backends), then use `teleport start --bootstrap` to restore the data to Postgres after configuring it.

**Reply by teenwolf99** [reply]
@webvictim The steps from the link was useful to move the cluster configuration (users, roles, etc) partial, but I was trying to move the users audit events from the fire store DB to PostgreSQL(like full migration).

**Reply by webvictim** [reply]
There's no tooling to do that unfortunately.
